chef/cookbooks/cinder/recipes/volume.rb

```diff
@@ -88,8 +88,16 @@ def make_volume(node, backend_id, volume)
 
   return if volume_exists(volname)
 
-  unclaimed_disks = BarclampLibrary::Barclamp::Inventory::Disk.unclaimed(node)
-  claimed_disks = BarclampLibrary::Barclamp::Inventory::Disk.claimed(node, "Cinder")
+  claim_string = "Cinder"
+
+  # do not claim any disk that has non-Cinder role assigned
+  disk_roles = node["crowbar_wall"].disk_roles || {}
+  unclaimed_disks = BarclampLibrary::Barclamp::Inventory::Disk.unclaimed(node).sort.select { |d|
+    role = disk_roles[d.unique_name] || ""
+    role.empty? || role == claim_string
```

Comment:
> shouldn't unclaimed() return only disks with role.empty?

We really want the disks that are unclaimed AND with Ceph role. Or with empty role, to simulate the old behavior when user doesn't have roles set